pkg/autoupdate/autoupdate.go

```diff
@@ -115,6 +116,8 @@ func ValidateImageReference(imageName string) error {
 // It returns a slice of successfully restarted systemd units and a slice of
 // errors encountered during auto update.
 func AutoUpdate(ctx context.Context, runtime *libpod.Runtime, options entities.AutoUpdateOptions) ([]*entities.AutoUpdateReport, []error) {
+	runtime.NewSystemEvent(events.AutoUpdate)
```

Comment:
I think we currently generated events after something was done. Right now this would report the auto-update event even when an error occurred. Should we just move it to the end of this function?

Reply:
I considered it at the end as well but decided to put it before. This way, I find the event logs easier to read.  As the events after can be very noisy I find it more useful to first know that auto-update was triggered:
```
auto update
pull
pull
rm
rm
create
start
create
start
...
```

Reply:
Can we place it after the systemd.ConnectToDBUS() call?

Reply:
Nothing which triggers other events should happen before that. What if ConnectToDBUS() fails? In this case auto-update didn't happen but we still created an event for it.

You listed events all happen inside autoUpdateRegistry()/autoUpdateLocally()

Reply:
Updated accordingly along with the tests.